kedro/extras/datasets/dask/parquet_dataset.py

```diff
@@ -105,8 +129,40 @@ def _load(self) -> dd.DataFrame:
         )
 
     def _save(self, data: dd.DataFrame) -> None:
+        self._process_schema()
         data.to_parquet(self._filepath, storage_options=self.fs_args, **self._save_args)
 
+    def _process_schema(self) -> None:
+        """This method processes the schema in the catalog.yml or the API, if provided.
+        This assumes that the schema is specified using Triad's grammar for
+        schema definition.
+
+        When the value of the `schema` variable is a string, it is assumed that
+        it corresponds to the full schema specification for the data.
+
+        Alternatively, if the `schema` is specified as a dictionary, then only the
+        columns that are specified will be strictly mapped to a field type. The other
+        unspecified columns, if present, will be inferred from the data.
+
+        This method converts the Triad-parsed schema into a pyarrow schema.
+        The output directly supports Dask's specifications for providing a schema
+        when saving to a parquet file.
+        """
+        schema = self._save_args.get("schema")
+
+        if isinstance(schema, dict):
+            pa_schema = triad.Schema(
+                ",".join([":".join(i) for i in schema.items()])
```

Comment:
You can construct `triad.Schema` directly from a dict; see https://github.com/fugue-project/triad/blob/master/triad/collections/schema.py#L59.

An example with string values:

```python
>>> Schema({"a": "int", "b": "float"}).pyarrow_schema
a: int32
b: float
```

I don't know that you can do more complex stuff, like nested lists, but I think it's a bit unnatural that you define the top level as a dict but the nested lists as strings (as in the test cases).

I think if you've got a power-user, they can just construct the schema string for more complex cases.

Reply:
Oh, I chose to implement it this way because of how `dask.to_parquet` distinguishes between a `dict`-like vs. a `pa.Schema`-like value for its `schema` argument.

If we directly cast a `dict` input into a schema as your snippet shows, `triad` would return a schema containing the `key:value` explicitly declared in the `dict` which would prevent dask from inferring the types for the other fields (if any). It could cause unexpected behaviors since there could be a case where a user simply wants to explicitly override the inferred field types for some of the columns and let dask do the inference for the rest.

In some sense, the `dict` input is a more general case since we can choose to specify the field types only for certain fields or for all the fields.

> it's a bit unnatural that you define the top level as a dict but the nested lists as strings (as in the test cases).
Ah, I was simulating how the parsed arguments from the `catalog.yml` would be passed into the dataset.

Since in the `catalog.yml`, the specification looks like:

```
save_args:
  schema:
    col1: int64
    col2: "[[int64]]"
```

then, I suppose that would be passed as a dict with this structure `{"col1": "int64", "col2": "[[int64]]"}`.